unidist/core/backends/mpi/core/controller/api.py

```diff
@@ -337,8 +353,12 @@ def put(data):
     unidist.core.backends.mpi.core.common.MasterDataID
         An ID of an object in object storage.
     """
-    data_id = object_store.generate_data_id(garbage_collector)
-    object_store.put(data_id, data)
+    local_store = LocalObjectStore.get_instance()
+    shared_store = SharedObjectStore.get_instance()
+    data_id = local_store.generate_data_id(garbage_collector)
+    local_store.put(data_id, data)
```

Comment:
No, we create the DataId here and only the current process can initiate clearing this data_id. If we call `get_unique_data_id` we will create a strong reference to this data_id and it will never be removed.